jscomp/frontend/bs_builtin_ppx.ml

```diff
@@ -228,15 +228,16 @@ let expr_mapper ~async_context ~in_function_def (self : mapper)
     let safe_module_type_lid : Ast_helper.lid =
       {txt = Lident (local_module_type_name txt); loc = me.pmod_loc}
     in
-    {
-      e with
-      pexp_desc =
-        Pexp_letmodule
-          ( lid,
-            Ast_await.create_await_module_expression
-              ~module_type_lid:safe_module_type_lid me,
-            expr );
-    }
+    self.expr self
```

Comment:
This is a little dangerous.
One never applies the recursion to the whole expression, but to the subparts.
The danger is to create infinite loops.

Reply:
Ah, agree.